### ISSUE TITLE
[documentation update] Individual page audit

### ISSUE THREAD

**Issue by chrisk-7777** [pmndrs/zustand#1220]
Core conversation here: https://github.com/pmndrs/zustand/discussions/1033

For each of the pages on the new documentation site, audit each page for clear, concise explanations, checking code snippets are correct, and conveys the correct information. 

Blocked by https://github.com/pmndrs/zustand/issues/1215

Current list of pages, may change based on 1215 (above)

 - [ ] 3rd party libraries
 - [ ] auto generating selectors
 - [ ] event handler in pre react 18
 - [ ] flux inspired practice
 - [ ] immutable state and merging
 - [ ] Update immutable state and merging
 - [ ] introduction
 - [ ] maps and sets usage
 - [ ] persisting store data
 - [ ] practice with no store actions
 - [ ] recipes
 - [ ] testing
 - [ ] typescript
 - [ ] updating nested state object values
 - [ ] v4 migration
 - [ ] zustand v3 create context

**Comment by Exalted100** [comment]
I'm working on subtask 'testing'.

**Comment by Exalted100** [comment]
Do I create a new PR for each subtask or should I create one PR when I'm done with all subtasks?

**Comment by chrisk-7777** [comment]
Some of the pages are quite huge (like recipes). 
Personally I would think smaller PR's are better - be it one or a few. 

@dai-shi ultimately will be the one approving/merging them, so I'll leave it up to him (he's away for a few days, so there may be a delay over the weekend).

**Comment by Exalted100** [comment]
I've worked on testing. Next, I'll work on autogenerating selectors, flux inspired practice, event handler in pre react 18, and immutable state and merging. I'll make a single PR for the changes I make to all five documents.

**Comment by Exalted100** [comment]
Made a PR for the completed tasks here: https://github.com/pmndrs/zustand/pull/1231

Next, will be working on maps and sets usage, practice with no store actions, typescript, and updating nested state object values. Of course it will be in a separate PR.

**Comment by dai-shi** [comment]
Yes, I will merge PRs. But if @chrisk-7777 approves PRs, it helps moving forward, so please feel free to do so. Also, I may explicitly request for reviews.

**Comment by chrisk-7777** [comment]
Happy to review PRs. @Exalted100 I'll review later tonight 👍

**Comment by chrisk-7777** [comment]
#1231 Approved. Well written @Exalted100

**Comment by sewera** [comment]
Hi @chrisk-7777 and @dai-shi!

I wanted to help with the documentation, but I wanted to ask you Chris to update the main comment if you find some time, so that it is clear which pages are left for audit.

I also can review some future doc PRs if some help is still necessary.

Cheers :)

**Comment by chrisk-7777** [comment]
@sewera can do. I've had to be away for the last few weeks, but aiming to get things back on track this week. 
I'll ping you with an update soon

**Comment by sewera** [comment]
Thanks a lot, looking forward to be a part of this <3

**Comment by sewera** [comment]
Hi, just dropping in with a quick reminder :)

**Comment by chrisk-7777** [comment]
Thanks @sewera !


As a starting point, can you review the integration section, including the following links:
 - https://docs.pmnd.rs/zustand/integrations/3rd-party-libraries
 - https://docs.pmnd.rs/zustand/integrations/updating-draft-states
 - https://docs.pmnd.rs/zustand/integrations/persisting-store-data

Key things to look for: 
 - General spelling and grammar
 - Double checking code snippets work
 - Double checking all links on page work, both in Github and on the docs site
 - A little more subjective, but a consistent "voice" on the page. Does it read well, and is it consistent with other pages. I say subjective because we don't have a clear definition of voice yet, but if anything obvious jumps out, please update it (or check in here :) )

In terms of working locally with the docs, I have a comment here that may help get started: https://github.com/pmndrs/zustand/discussions/1033#discussioncomment-3967766

Feel free to split the above into 3 PR's or one, whatever is easiest for you. 
Also feel free to "@" me anytime and I'll make sure to respond here

**Comment by sewera** [comment]
Thanks for the reply, really appreciate your guide, I'll start right away!

**Comment by sewera** [comment]
I managed to revise only the third-party libraries documentation page today. I opened a PR for it, so that you can start reviewing it right away @chrisk-7777 :)

**Comment by sewera** [comment]
Hi @chrisk-7777, I saw that you've already been notified, but I'll gather all the information here:

- I've opened two PRs:
  - The first for the [Third-party libraries page](https://github.com/pmndrs/zustand/pull/1478), and
  - the second for the [Immer middleware page](https://github.com/pmndrs/zustand/pull/1479);
- I've added a supplementary comment for the [local docs dev setup](https://github.com/pmndrs/zustand/discussions/1033#discussioncomment-4399876);
- Unfortunately, I won't be able to take up the third page (`persisting-store-data.md`/Persist middleware) for revision anytime soon. If there is no volunteer, I'll maybe find some time next week or later;
- Obviously, I'll maintain my PRs and act on all the feedback I get there.

Thanks for your help, it's a pleasure to collaborate with you :)

**Comment by sewera** [comment]
Hi @chrisk-7777 and @dai-shi!
I found some time to revise `persisting-store-data.md`,
as I mentioned in [my previous comment](https://github.com/pmndrs/zustand/issues/1220#issuecomment-1351315903),
and I've opened a PR for it — #1497.

Please review it when you are free.
I'll maintain my PR accordingly.

Happy to be a part of it, cheers! :)

**Comment by sewera** [comment]
Thank you guys for collaboration and PR reviews, it's very fun working with you!

**Comment by sewera** [comment]
Hi @chrisk-7777 and @dai-shi!

The time has come for "v4 migrations" page.

The PR is #1507.

As usual, please review it when you are free.
I'll maintain my PR accordingly.

**Comment by chrisk-7777** [comment]
Thanks @sewera , really well written and explained. 

I haven't really been keeping on top of the docs audit and rollout as much as I would've liked. Mainly due to work commitments. 

If it's ok with Dai Shi, are you interested in becoming more of the "go to" person for the docs? I feel like you have a really good eye for detail and the overarching structure.

**Comment by sewera** [comment]
Sure, @chrisk-7777. However, I'm only available on Wednesdays, and I can dedicate up to 2 hours weekly. If it's fine with you, it's also fine with me.

CC: @dai-shi

**Comment by dai-shi** [comment]
Sounds reasonable. Let's cooperate together.
@sewera Just sent an invitation.

**Comment by sewera** [comment]
From what I've seen so far, the documentation on the Pmndrs website
is somewhat stable by now. If there are no actions left in this issue,
I'd say we should close it. What do you think @dai-shi and @chrisk-7777?

**Comment by chrisk-7777** [comment]
I feel like the sites docs are now in a good enough state that we can close this issue, and continue to iterate on the docs as needed. 
Not to say the docs are complete, but I don't think it warrants an open ongoing issue anymore, rather smaller focused issues as they crop up

**Comment by sewera** [comment]
That's exactly what I meant, I'm happy that we agree.
Amazing chapter in my open-source journey, so thank you all for that so far :)

**Comment by dai-shi** [comment]
Thanks you all for your contributions!
